cluster_tools/cluster_tools/schedulers/slurm.py

```diff
@@ -210,10 +210,14 @@ def inner_handle_kill(self, *args, **kwargs):
         ]
 
         if len(scheduled_job_ids):
-            # Array jobs (whose id looks like `<job_id>_<array_index>`) don't need to be canceled individually,
-            # but can be canceled together using the job_id
+            # Array jobs (whose id looks like `<job_id>_<array_index>`) don't need to be signaled individually,
+            # but can be canceled together using the job_id.
             unique_job_ids = set(map(lambda x: str(x).split("_")[0], scheduled_job_ids))
-            _stdout, stderr, exit_code = call(f"scancel {' '.join(unique_job_ids)}")
+            # Send SIGINT signal to running jobs instead of terminating the jobs right away. This way, the jobs can
+            # react to the signal, safely shutdown and signal (cancel) jobs they possibly scheduled, recursively.
+            _stdout, stderr, exit_code = call(
+                f"scancel --state=PENDING {' '.join(unique_job_ids)}; scancel -s SIGINT --state=RUNNING {' '.join(unique_job_ids)}"
```

Comment:
According to the docs of [`scancel`](https://slurm.schedmd.com/scancel.html) there can also be `SUSPENDED` jobs. Maybe also send the default signal to these jobs?

Reply:
We discussed this already for the not-recursive case, but I want to doublecheck again: If a job doesn't react to SIGINT, there is no follow-up action taken here, right? Since slurm jobs were not canceled automatically at all in the past, it's probably okay to go with this behavior (let's see whether it occurs at all), right?

Reply:
I'd say yes to both. Generally, slurm jobs are like separate voxelytics instances. So if the shutdown procedure works for the sequential and multiprocessing strategies, and the SIGINT signal propagation to the slurm job works, the shutdown procedure for slurm jobs should work as well.